### PR TITLE
Fix scrollbar configuration for hover events

### LINKED ISSUE #72814: [web] Scrollbar creating exception by gesture library - (Reference: Scrollbar Fixes & Updates #70866)
## Steps to Reproduce by using Flutter Web

1. Add a Scrollbar to the Scaffold Body using a ScrollController
2. Run app and when you move your mouse inside the body you received the error 'Exception caught by gesture library' (I have not added a Gesture widget)

Source Code:
```
ScrollController _scrollController = ScrollController();

body: Scrollbar(
  isAlwaysShown: true,
  thickness: 10.0,
  controller: _scrollController,
  child: SingleChildScrollView(
    controller: _scrollController,
    child: Form(
      key: _formStateKey,
      onChanged: _onFormChanged,
      onWillPop: _showSaveWarningDialog,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: DataEntryContainer(),
      ),
    ),
  ),
),
```

**Expected results:** 
Should not get errors.

**Actual results:** 
```
======== Exception caught by gesture library =======================================================
The following assertion was thrown while dispatching a pointer event:
Assertion failed: file:///C:/Users/marco/flutter/packages/flutter/lib/src/material/scrollbar.dart:246:12
!_isMobile
is not true

When the exception was thrown, this was the stack: 
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 231:49  throw_
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3    assertFailed
packages/flutter/src/material/scrollbar.dart 246:13                                                                        handleHover
packages/flutter/src/rendering/proxy_box.dart 2906:14                                                                      handleEvent
packages/flutter/src/gestures/binding.dart 361:14                                                                          dispatchEvent
...
Event: PointerHoverEvent#2b67a(position: Offset(608.0, 175.4))
  position: Offset(608.0, 175.4)
Target: RenderMouseRegion#213e7 relayoutBoundary=up23
  needs compositing
  parentData: <none> (can use size)
  constraints: BoxConstraints(0.0<=w<=540.0, 0.0<=h<=484.0)
  size: Size(540.0, 433.1)
  listeners: hover, exit
====================================================================================================
```

<details>
  <summary>Logs</summary>

flutter run --verbose
```
[   +7 ms] [CHROME]:DevTools listening on ws://127.0.0.1:60705/devtools/browser/06b61ea9-f6c9-4d93-9ba8-aae8727c8d64
[+1504 ms] DwdsInjector: Received request for entrypoint at http://localhost:60671/main_module.bootstrap.js
[   +4 ms] MetadataProvider: Loading debug metadata...
[   +9 ms] MetadataProvider: Loaded debug metadata
[   +8 ms] DwdsInjector: Injected debugging metadata for entrypoint at http://localhost:60671/main_module.bootstrap.js
[+5650 ms] DevHandler: Debug service listening on ws://127.0.0.1:60751/FRb6Wc-JgjI=/ws

[  +10 ms] Debug service listening on ws://127.0.0.1:60751/FRb6Wc-JgjI=/ws
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[   +1 ms] Warning: Flutter's support for web development is not stable yet and hasn't
[        ] been thoroughly tested in production environments.
[        ] For more information see https://flutter.dev/web
[        ]   To hot restart changes while running, press "r" or "R".
[   +1 ms] For a more detailed help message, press "h". To quit, press "q".
[+11707 ms] LIST: 9
[+4298 ms] ══╡ EXCEPTION CAUGHT BY GESTURE LIBRARY ╞═══════════════════════════════════════════════════════════
[   +2 ms] The following assertion was thrown while dispatching a pointer event:
[   +1 ms] Assertion failed:
[        ] file:///C:/Users/marco/flutter/packages/flutter/lib/src/material/scrollbar.dart:246:12
[        ] !_isMobile
[        ] is not true
[        ] When the exception was thrown, this was the stack:
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 231:49  throw_
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 24:3    assertFailed
[        ] packages/flutter/src/material/scrollbar.dart 246:13                                                                        handleHover
[        ] packages/flutter/src/rendering/proxy_box.dart 2906:14                                                                      handleEvent
[        ] packages/flutter/src/gestures/binding.dart 361:14                                                                          dispatchEvent
[   +1 ms] packages/flutter/src/rendering/binding.dart 279:11                                                                         dispatchEvent
[        ] packages/flutter/src/gestures/binding.dart 316:7                                                                           [_handlePointerEventImmediately]
[   +1 ms] packages/flutter/src/gestures/binding.dart 280:5                                                                           handlePointerEvent
[        ] packages/flutter/src/gestures/binding.dart 238:7                                                                           [_flushPointerEventQueue]
[        ] packages/flutter/src/gestures/binding.dart 221:32                                                                          [_handlePointerDataPacket]
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart 911:13            invoke1
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/platform_dispatcher.dart 170:5             invokeOnPointerDataPacket
[   +2 ms] C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/pointer_binding.dart 128:39                [_onPointerData]
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/pointer_binding.dart 552:16                <fn>
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/pointer_binding.dart 506:21                <fn>
[        ] C:/b/s/w/ir/cache/builder/src/out/host_debug/flutter_web_sdk/lib/_engine/engine/pointer_binding.dart 214:16                <fn>
[        ] Event:
[        ]   PointerHoverEvent#4e217(position: Offset(766.9, 244.6))
[        ] Target:
[        ]   RenderMouseRegion#418b7 relayoutBoundary=up23 NEEDS-PAINT
[        ] ════════════════════════════════════════════════════════════════════════════════════════════════════
[
```

flutter doctor -v
NOTE: I am not testing Flutter Windows and Visual Studio is not installed. I am testing Flutter Web.
```
[√] Flutter (Channel master, 1.26.0-2.0.pre.103, on Microsoft Windows [Version 10.0.19042.685], locale en-US)
    • Flutter version 1.26.0-2.0.pre.103 at C:\Users\marco\flutter
    • Framework revision 32741c0ec8 (17 hours ago), 2020-12-21 18:34:02 -0600
    • Engine revision af6889a600
    • Dart version 2.12.0 (build 2.12.0-169.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\marco\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.52.1)
    • VS Code at C:\Users\marco\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.685]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 87.0.664.66

! Doctor found issues in 1 category.
```

</details>

Thank you.

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This fixes the scrollbar configuration for hover events. They should only be triggered when the PointerDeviceKind is a mouse, hover events triggered by a tap for example should be ignored. This is because tapping on the scrollbar would otherwise trigger a hover enter event, at which point there would be no way to trigger a hover exit - which the scrollbar is dependent on.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/72814
https://github.com/flutter/flutter/issues/70866

## Tests

Updated hover trigger test to verify pointer device kind.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by darrenaustin — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/scrollbar.dart [hunk 1]
```diff
@@ -174,7 +171,7 @@ class _ScrollbarState extends RawScrollbarState<Scrollbar> {
       if (states.contains(MaterialState.hovered) && widget.showTrackOnHover)
         return widget.hoverThickness ?? _kScrollbarThicknessWithTrack;
       // The default scrollbar thickness is smaller on mobile.
-      return widget.thickness ?? (_kScrollbarThickness / (_isMobile ? 2 : 1));
+      return widget.thickness ?? (_kScrollbarThickness / (_useAndroidScrollbar ? 2 : 1));
```
Comment (xu-baolin) [inline packages/flutter/lib/src/material/scrollbar.dart:174]: This change the iPhone behavior?

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 2]
```diff
@@ -1165,27 +1152,42 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
...
+              switch(event.kind) {
+                case PointerDeviceKind.mouse:
+                  handleHoverExit(event);
+                  break;
+                case PointerDeviceKind.stylus:
+                case PointerDeviceKind.invertedStylus:
+                case PointerDeviceKind.unknown:
+                case PointerDeviceKind.touch:
+                  break;
+              }
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1171]: Why not,
```dart
if (event.kind == PointerDeviceKind.mouse)
  handleHoverExit(event);
```

File: packages/flutter/lib/src/material/scrollbar.dart (hunk 1 above)
Comment (xu-baolin) [reply packages/flutter/lib/src/material/scrollbar.dart:174]: We can just remove the override `didChangeDependencies` and provide a method,
```dart
  double _getDefaultThickness() {
    final ThemeData theme = Theme.of(context);
    switch (theme.platform) {
      case TargetPlatform.android:
        return _kScrollbarThickness;
      default:
        return _kScrollbarThickness / 2;
    }
  }
```

Comment (xu-baolin) [reply packages/flutter/lib/src/material/scrollbar.dart:174]: > This change the iPhone behavior?

yep, the `CupertinoScrollbar` override this!

Comment (Piinks) [reply packages/flutter/lib/src/material/scrollbar.dart:174]: It does, but I have another change on the way to address that. Now that the refactor to RawScrollbar is complete, I am going to follow up to restore the adaptive behavior that returns the CupertinoScrollbar on iOS. :)

File: packages/flutter/lib/src/widgets/scrollbar.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1171]: Cleaner for sure! I think we usually use switches for enums, it makes sure we don't miss a case in future changes. https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values

**Pull request merged**